dgl_ptm/dgl_ptm/model/initialize_model.py

```diff
@@ -392,11 +394,27 @@ def step(self):
             #TODO add model dump here. Also check against previous save to avoid overwriting
             raise RuntimeError(f'execution of step failed for step {self.step_count}')
 
-    def run(self):
-        """ run the model for each step until the step_target is reached."""
+    def run(self, restart=False):
+        """
+        run the model for each step until the step_target is reached.
+
+        param: restart: boolean or int or a pair of ints, optional.
+        If True, the model is run from last checkpoint,
```

Comment:
We should be able to identify the `"last checkpoint"` by a `"milestone"` and `"step"` as (milestone, step), correct? This can be merged with the third case to simplify the code.

Reply:
Based on our discussion, there was some confusion here. The tuple case would be (step, instance), not (milestone, step). I'll try to clarify this in the doc string.